compiler/src/iree/compiler/Dialect/HAL/Target/TargetRegistry.cpp

```diff
@@ -14,54 +14,82 @@ namespace IREE {
 namespace HAL {
 
 // Returns the static registry of translator names to translation functions.
-static llvm::StringMap<TargetBackendRegistration *>
-    &getMutableTargetRegistry() {
-  static llvm::StringMap<TargetBackendRegistration *> registry;
-  return registry;
+static TargetBackendRegistry &getMutableTargetRegistry() {
+  static TargetBackendRegistry global;
+  return global;
+}
+
+const TargetBackendRegistry &TargetBackendRegistry::getGlobal() {
+  return getMutableTargetRegistry();
 }
 
 TargetBackendRegistration::TargetBackendRegistration(llvm::StringRef name,
-                                                     CreateTargetBackendFn fn)
+                                                     CreateTargetBackendFn fn,
+                                                     bool registerStaticGlobal)
     : initFn(std::move(fn)) {
-  auto &registry = getMutableTargetRegistry();
-  if (registry.count(name) > 0) {
-    llvm::report_fatal_error(
-        "Attempting to overwrite an existing translation backend");
+  if (registerStaticGlobal) {
+    auto &registry = getMutableTargetRegistry();
+    if (registry.registrations.count(name) > 0) {
```

Comment:
`StringMap` has a contains method, FYI: https://llvm.org/doxygen/classllvm_1_1StringMap.html#a5bdab9e85b82bae5a3470d4fa0cf574c

and I think LLVM has a contains method for generic containers in its STL